src/beanmachine/ppl/inference/predictive.py

```diff
@@ -79,7 +81,14 @@ def simulate(  # noqa: C901
                 for rvid, rv in query_dict.items():
                     if rv.dim() > 2:
                         query_dict[rvid] = rv.squeeze(0)
-                return MonteCarloSamples(query_dict)
+                post_pred = MonteCarloSamples(query_dict)
+                if return_inference_data:
+                    post = posterior.to_inference_data()
+                    post_pred = post_pred.to_inference_data().posterior
```

Comment:
This diff stack is introducing 1 typecheck errors. Please fix these before landing. You may run Pyre locally in fbcode with `pyre -l beanmachine/beanmachine/ppl`.

[beanmachine/beanmachine/ppl/inference/predictive.py:87:32] Undefined attribute [16]: `az.data.inference_data.InferenceData` has no attribute `posterior`.

Reply:
:) I ran into a similar issue when working with `MonteCarloSamples` and ended up [throwing a `#pyre-ignore` there](https://github.com/facebookresearch/beanmachine/blob/2e99c73fd25b6a68d680a60a6aeb8d15171e58d3/src/beanmachine/ppl/inference/monte_carlo_samples.py#L162-L163). You may also consider just using `post_pred.to_xarray()` since the method basically does the same thing under the hood.